### PR TITLE
Struts 7.0.x

### PR DESCRIPTION
**Pull request by lukaszlenart**
SNAPSHOT build is available in ASF [Snapshot repository](https://struts.apache.org/builds.html#maven-snapshots) or you can download Nightlies build from [here](https://nightlies.apache.org/struts/snapshot/)

### CODE REVIEW

**Comment by jdyer1** [comment]
@lukaszlenart  Is there anything I can do to help us with completing the last mile and getting that 7.0 M1 release out the door?  

For my part, I have used the latest snapshot release and have migrated my company's Big Struts Application and we have not yet found any problems. I will say, however, we only use a small subset of the features.  For reference I am evaluating with Wildfly 31-Beta.

**Comment by lukaszlenart** [comment]
@jdyer1 sorry, I was distracted by work on Struts 6.4.0, I'm just preparing a reverse merge of the master branch and I will issue the first M release ASAP :)

**Comment by lukaszlenart** [comment]
Done, [Struts 7.0.0-M1](https://twitter.com/TheApacheStruts/status/1748665440425976009) is under test :)

File: core/pom.xml [hunk 1]
```diff
@@ -214,14 +215,14 @@
 
         <!-- File upload -->
         <dependency>
-            <groupId>commons-fileupload</groupId>
-            <artifactId>commons-fileupload</artifactId>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-fileupload2-jakarta</artifactId>
+            <version>2.0.0-M1</version>
```
Comment (deki) [inline core/pom.xml:220]: Can you please use
```
        <dependency>
            <groupId>org.apache.commons</groupId>
            <artifactId>commons-fileupload2-jakarta-servlet6</artifactId>
            <version>2.0.0-M2</version>
        </dependency>
```
instead?

File: pom.xml [hunk 2]
```diff
@@ -864,9 +817,9 @@
                 <version>4.4</version>
             </dependency>
             <dependency>
-                <groupId>commons-fileupload</groupId>
-                <artifactId>commons-fileupload</artifactId>
-                <version>1.5</version>
+                <groupId>org.apache.commons</groupId>
+                <artifactId>commons-fileupload2-jakarta</artifactId>
+                <version>2.0.0-M1</version>
```
Comment (deki) [inline pom.xml:822]: Can you please use
```
        <dependency>
            <groupId>org.apache.commons</groupId>
            <artifactId>commons-fileupload2-jakarta-servlet6</artifactId>
            <version>2.0.0-M2</version>
        </dependency>
```
instead?

**Review by deki — CHANGES_REQUESTED** [review]

File: core/pom.xml (hunk 1 above)
Comment (lukaszlenart) [reply core/pom.xml:220]: Feel free to open a PR :)

Comment (lukaszlenart) [reply core/pom.xml:220]: In progress https://github.com/apache/struts/pull/861

Comment (deki) [reply core/pom.xml:220]: Thanks Lukasz, I still had it on my list for this week.

**Comment by jellisgwn** [comment]
hi! 
Is there a targeted release date for 7.0.0? Are there known issues still to be resolved?

**Comment by lukaszlenart** [comment]
> hi! Is there a targeted release date for 7.0.0? Are there known issues still to be resolved?

No exact date, first Struts 2.5.x must become EOL which will happen in a month

**Comment by jellisgwn** [comment]
> > hi! Is there a targeted release date for 7.0.0? Are there known issues still to be resolved?
> 
> No exact date, first Struts 2.5.x must become EOL which will happen in a month

ah, good to know. wasn't aware of that requirement.

**Comment by ctmay4** [comment]
I realize there is no estimated date for the final 7.0 version. However how close to the final version is the M6 release? We are stuck on a very old version of Wildfly because the later versions switched to jakarta so we are stuck until we can update Struts. I'm thinking about releasing with a milestone release of Struts if it is close the final.

**Comment by lukaszlenart** [comment]
@ctmay4 see discussion on [the dev mailing list](https://lists.apache.org/thread/5b8nytjl0dsnwkzw5q4yk6x4wmrq44ly)

**Comment by JulB310** [comment]
Hi, is there any release date planned for stable version of Struts 7 ? Thanks

**Comment by lukaszlenart** [comment]
> Hi, is there any release date planned for stable version of Struts 7 ? Thanks

A rough estimate is about mid/end of September

**Pull request merged**